libandroid-navigation/src/commonMain/kotlin/org/maplibre/navigation/core/location/Location.kt

```diff
@@ -0,0 +1,57 @@
+package org.maplibre.navigation.core.location
+
+import android.os.SystemClock
+import kotlinx.datetime.Clock
+import kotlinx.datetime.DateTimeUnit
+import kotlinx.datetime.Instant
+import org.maplibre.geojson.Point
+
+/**
+ * A generic model that represents a user location.
+ */
+data class Location(
+    /**
+     * Latitude, in degrees.
+     */
+    val latitude: Double,
+
+    /**
+     * Longitude, in degrees.
+     */
+    val longitude: Double,
+
+    /**
+     * Horizontal accuracy of the latitude and longitude, in meters.
+     * If not available, it will be `null`.
+     */
+    val accuracyMeters: Float? = null,
+
+    /**
+     * Speed of user in meters per second.
+     * If not available, it will be `null`.
+     */
+    val speedMetersPerSeconds: Float? = null,
+
+    /**
+     * Bearing of the user, in degrees.
+     * If not available, it will be `null`.
+     */
+    val bearing: Float? = null,
+
+    /**
+     * Date & time of this location fix in UTC zone.
+     */
+    val time: Instant = Clock.System.now(),
```

Comment:
We made some bad experiences with kotlinx.datetime, especially since they have issues with Android SDK 26 and below:

> If you target Android devices running below API 26, you need to use Android Gradle plugin 4.0 or newer and enable core library desugaring.

Not sure if you did that, but in general, we could just use time millis (or actually time in seconds - because that's what we get from iOS) and simply store this here? So no instant, simple Long?

Reply:
Wow okay.. Okay then I would also suggest to use Long as time. I'm more fan of using objects instead of natives. But I'm on your side in this case. I will change this